File: _includes/code/java-v6/src/test/java/quickstart/QuickstartLocalCreate.java
package quickstart;

// START CreateCollection
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.Property;
import io.weaviate.client6.v1.api.collections.VectorConfig;
import io.weaviate.client6.v1.api.collections.WeaviateObject;
import io.weaviate.client6.v1.api.collections.batch.BatchContext;

import java.util.List;
import java.util.Map;

public class QuickstartLocalCreate {

  public static void main(String[] args) throws Exception {
    WeaviateClient client = null;
    String collectionName = "Movie";

    try {
      // Step 1.1: Connect to your local Weaviate instance
      client = WeaviateClient.connectToLocal();
      // END CreateCollection

      // NOT SHOWN TO THE USER - DELETE EXISTING COLLECTION
      if (client.collections.exists(collectionName)) {
        client.collections.delete(collectionName);
      }

      // START CreateCollection
      // Step 1.2: Create a collection
      // highlight-start
      client.collections.create(collectionName,
          col -> col.vectorConfig(VectorConfig.text2vecOllama(v -> v
              // If using Docker you might need: http://host.docker.internal:11434
              .apiEndpoint("http://ollama:11434")
              .model("nomic-embed-text") // The model to use
          ))
              // Define properties for the collection
              .properties(Property.text("title"), Property.text("description"),
                  Property.text("genre")));
      // highlight-end

      // Step 1.3: Import three objects
      List<Map<String, Object>> dataObjects = List.of(Map.of("title",
          "The Matrix", "description",
          "A computer hacker learns about the true nature of reality and his role in the war against its controllers.",
          "genre", "Science Fiction"),
          Map.of("title", "Spirited Away", "description",
              "A young girl becomes trapped in a mysterious world of spirits and must find a way to save her parents and return home.",
              "genre", "Animation"),
          Map.of("title", "The Lord of the Rings: The Fellowship of the Ring",
              "description",
              "A meek Hobbit and his companions set out on a perilous journey to destroy a powerful ring and save Middle-earth.",
              "genre", "Fantasy"));

      // Insert the objects using server-side batching
      CollectionHandle<Map<String, Object>> movies =
          client.collections.use(collectionName);
      BatchContext<Map<String, Object>> batch = movies.batch.start();
      // Closing the batch sends the remaining objects and waits for the results
      try (batch) {
        for (Map<String, Object> properties : dataObjects) {
          batch.add(
              WeaviateObject.<Map<String, Object>>of(o -> o.properties(properties)));
        }
      }

      if (batch.numberOfErrors() > 0) {
        System.err
            .println("Number of failed imports: " + batch.numberOfErrors());
      } else {
        System.out.println("Imported & vectorized " + dataObjects.size()
            + " objects into the Movie collection");
      }
    } finally {
      if (client != null) {
        client.close(); // Free up resources
      }
    }
  }
}
// END CreateCollection

File: _includes/code/csharp/quickstart/QuickstartLocalCreate.cs
// START CreateCollection
using Weaviate.Client;
using Weaviate.Client.Models;
using System;
using System.Threading.Tasks;
using System.Collections.Generic;
using System.Threading;

namespace WeaviateProject.Examples
{
    public class QuickstartLocalCreate
    {
        public static async Task Run()
        {
            string collectionName = "Movie";

            // Step 1.1: Connect to your local Weaviate instance
            var client = await Connect.Local();
            // END CreateCollection
            // NOT SHOWN TO THE USER - DELETE EXISTING COLLECTION
            if (await client.Collections.Exists(collectionName))
            {
                await client.Collections.Delete(collectionName);
            }
            // START CreateCollection

            // Step 1.2: Create a collection
            var movies = await client.Collections.Create(new CollectionConfig
            {
                Name = collectionName,
                VectorConfig = Configure.Vectors.Text2VecOllama(
                    apiEndpoint: "http://ollama:11434", model: "nomic-embed-text"
                ).New(name: "default"),
                // Define properties for the collection
                Properties =
                [
                    Property.Text("title"),
                    Property.Text("description"),
                    Property.Text("genre")
                ]
            });

            // Step 1.3: Import three objects
            var dataObjects = new List<object>
            {
                new {
                    title = "The Matrix",
                    description = "A computer hacker learns about the true nature of reality and his role in the war against its controllers.",
                    genre = "Science Fiction"
                },
                new {
                    title = "Spirited Away",
                    description = "A young girl becomes trapped in a mysterious world of spirits and must find a way to save her parents and return home.",
                    genre = "Animation"
                },
                new {
                    title = "The Lord of the Rings: The Fellowship of the Ring",
                    description = "A meek Hobbit and his companions set out on a perilous journey to destroy a powerful ring and save Middle-earth.",
                    genre = "Fantasy"
                }
            };

            // Insert objects using InsertMany
            var insertResponse = await movies.Data.InsertMany(dataObjects.ToArray());

            if (insertResponse.HasErrors)
            {
                Console.WriteLine($"Errors during import: {insertResponse.Errors}");
            }
            else
            {
                Console.WriteLine($"Imported & vectorized {insertResponse.Count} objects into the Movie collection");
            }
            // END CreateCollection
            Thread.Sleep(1000);
            // START CreateCollection
        }
    }
}
// END CreateCollection